### PR TITLE
[Term Entry] Dart List .insert()

### LINKED ISSUE #4309: [Term Entry] Dart List .insert()
### Reason/inspiration (optional)

We would like a new term entry under Dart [List](https://www.codecademy.com/resources/docs/dart/concept/list) for the Dart List `.insert()` method. 
Provide the entry in a markdown file under `docs/content/dart/concept/list/terms/insert/insert.md`

The entry should
    - Have an introductory paragraph about what the method is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the method in use.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by avdhoottt**
### Issue Solved

Closes #4309 

### Type of Change

- Adding a new entry

### Checklist

- [ ] All writings are my own.
- [ ] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [ ] I have performed a self-review of my own writing and code.
- [ ] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/dart/concepts/list/terms/insert/insert.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+Title: '.insert()'
+Description: 'Inserts multiple elements into a list at a specified index.'
```
Comment (letsbuildanything) [inline content/dart/concepts/list/terms/insert/insert.md:3]:
```suggestion
Description: 'Inserts an element at a specified index in a list.'
```

File: content/dart/concepts/list/terms/insert/insert.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+```pseudo
+list.insert(int index, E element);
+```
+
+- `index`: The position where the element will be inserted.
+- `element`: The element is the value to be added.
+
+## Example
+
+In the following example, the `.insert()` method is used to insert the number 3 at the second index:
```
Comment (letsbuildanything) [inline content/dart/concepts/list/terms/insert/insert.md:30]:
```suggestion
In the following example, the `.insert()` method is used to insert the number `3` at the second index:
```

File: content/dart/concepts/list/terms/insert/insert.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  List<int> numbers = [1, 2, 4, 5];
+  print('Original list: $numbers');
+  
+  // Inserting the number 3 at index 2
+  numbers.insert(2, 3);
+  print('List after insertion: $numbers');
+}
+```
+
+The output of the above code is as follows:
```
Comment (letsbuildanything) [inline content/dart/concepts/list/terms/insert/insert.md:43]:
```suggestion
The output of the above example code is as follows:
```

**Review by letsbuildanything — CHANGES_REQUESTED** [review]
@avdhoottt 
Apologies for the delay in reviewing this!
I've requested a few minor changes.

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @avdhoottt 😄 

The entry is ready to be merged! 🚀

**Pull request merged**